### PR TITLE
Tor works

### PR DESCRIPTION
**Pull request by gabidi**
# Description

- Embeds Tor Daemon + Onion Routing into Zeus, removing the dependency on Orbot and allowing Tor usage on IOS
- Updates backend APIs to use Tor when enabled

Relates to issue: **ZEUS-166**
Claims iOS + Android Tor Bounty:
https://github.com/ZeusLN/zeus/blob/master/docs/Bounties.md#tor-on-android
https://github.com/ZeusLN/zeus/blob/master/docs/Bounties.md#tor-on-ios

_Please enter a description and screenshots, if appropriate, of the work covered in this PR_

This pull request is categorized as a:

- [x] New feature
- [ ] Bug fix
- [ ] Code refactor
- [x] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [x] I’ve run `npm run tsc` and make sure my code didn’t produce errors 
- [x] I’ve run `npm run prettier` and formatted my code correctly

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [x] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [x] LND
- [x] c-lightning-REST
- [ ] spark
- [x] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [x] Contributors will need to run `npm install` after this PR is merged in
- [x] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [x] Changes were made that require an update to onboarding

### CODE REVIEW

**Comment by AliMeer** [comment]
Looks great so far 👍
Where are tor daemon settings configured?

**Comment by ZeusLN** [comment]
`enableTor` should be found on lines 188, 203, 257 (roughly) of `AddEditNode.tsx` to ensure the property is loaded properly on the edit node screen

**Review by MONIMAKER365 — APPROVED** [review]

**Review by MONIMAKER365 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "react-native-secure-key-store": "^2.0.5",
     "react-native-svg": "^9.13.6",
     "react-native-tcp": "^3.3.2",
+    "react-native-tor": "github:Sifir-io/react-native-tor",
```
Comment (kaloudis) [inline package.json:65]: Can you lock in the version here now that the package is public?

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -1,6 +1,4 @@
 module.exports = {
   root: true,
   extends: '@react-native-community',
-  comma-dangle: 0,
-  radix: 'as-needed',
```
Comment (MONIMAKER365) [inline .eslintrc.js:5]: `  comma-dangle: 0,`
  radix: 'as-needed',

**Review by MONIMAKER365 — CHANGES_REQUESTED** [review]

**Review by MONIMAKER365 — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (gabidi) [reply package.json]: locked to npm version 0.1.0

**Comment by kaloudis** [comment]
Looks good to merge if you tackle https://github.com/ZeusLN/zeus/pull/394#issuecomment-741945580

**Comment by gabidi** [comment]
> Looks good to merge if you tackle [#394 (comment)](https://github.com/ZeusLN/zeus/pull/394#issuecomment-741945580)

Didn't see that one before. Not sure what it means but will check and let you know.

**Review by MONIMAKER365 — APPROVED** [review]

**Review by MONIMAKER365 — APPROVED** [review]

**Review by MONIMAKER365 — DISMISSED** [review]

**Review by MONIMAKER365 — APPROVED** [review]

**Review by MONIMAKER365 — APPROVED** [review]
approved

**Review by kaloudis — DISMISSED** [review]

**Pull request merged**